src/Data/Swagger/Internal.hs

```diff
@@ -650,55 +718,50 @@ type HttpStatusCode = Int
 -- | Describes a single response from an API Operation.
 data Response = Response
   { -- | A short description of the response.
-    -- GFM syntax can be used for rich text representation.
+    -- [CommonMark syntax](https://spec.commonmark.org/) can be used for rich text representation.
     _responseDescription :: Text
 
-    -- | A definition of the response structure.
-    -- It can be a primitive, an array or an object.
-    -- If this field does not exist, it means no content is returned as part of the response.
-    -- As an extension to the Schema Object, its root type value may also be "file".
-    -- This SHOULD be accompanied by a relevant produces mime-type.
-  , _responseSchema :: Maybe (Referenced Schema)
+    -- | A map containing descriptions of potential response payloads.
+    -- The key is a media type or media type range and the value describes it.
+    -- For responses that match multiple keys, only the most specific key is applicable.
+    -- e.g. @text/plain@ overrides @text/*@.
+  , _responseContent :: InsOrdHashMap MediaType MediaTypeObject
 
-    -- | A list of headers that are sent with the response.
-  , _responseHeaders :: InsOrdHashMap HeaderName Header
+    -- | Maps a header name to its definition.
+  , _responseHeaders :: InsOrdHashMap HeaderName (Referenced Header)
 
-    -- | An example of the response message.
-  , _responseExamples :: Maybe Example
+    -- | A map of operations links that can be followed from the response.
+    -- The key of the map is a short name for the link, following the naming
+    -- constraints of the names for 'Component' Objects.
+  , _responseLinks :: InsOrdHashMap Text (Referenced Link)
   } deriving (Eq, Show, Generic, Data, Typeable)
 
 instance IsString Response where
-  fromString s = Response (fromString s) Nothing mempty Nothing
+  fromString s = Response (fromString s) mempty mempty mempty
+
+-- | A map of possible out-of band callbacks related to the parent operation.
+-- Each value in the map is a 'PathItem' Object that describes a set of requests that
+-- may be initiated by the API provider and the expected responses.
+-- The key value used to identify the path item object is an expression, evaluated at runtime,
+-- that identifies a URL to use for the callback operation.
+newtype Callback = Callback (InsOrdHashMap Text PathItem)
+  deriving (Eq, Show, Generic, Data, Typeable)
 
 type HeaderName = Text
 
+-- TODO this is mostly a copy of 'Param'.
 data Header = Header
   { -- | A short description of the header.
-    _headerDescription :: Maybe Text
+    _headerDescription :: Maybe HeaderName
 
-  , _headerParamSchema :: ParamSchema ('SwaggerKindNormal Header)
+  , _headerSchema :: Maybe (Referenced Schema)
   } deriving (Eq, Show, Generic, Data, Typeable)
```

Comment:
From the [OpenAPI v3.0.3 specification](http://spec.openapis.org/oas/v3.0.3#header-object).

> The Header Object follows the structure of the Parameter Object with the following changes:
> 
>  name MUST NOT be specified, it is given in the corresponding headers map.
>  in MUST NOT be specified, it is implicitly in header.
>  All traits that are affected by the location MUST be applicable to a location of header (for example, style).

The following properties seem to be missing:

- `required`
- `deprecated`
- `allowEmptyValue`
- `style`
- `explode`
- `example`
- `examples`

A comment shall be added if these properties are not supported.